util/devel/greptests

```diff
@@ -1,13 +1,82 @@
 #!/usr/bin/env bash
 
+if [ $# -eq 0 ]
+  then
+      echo "No arguments supplied, see -help"
+      exit 1
+fi
+
+if [ $# -gt 3 ]
+  then
+      echo "Only usable with single strings, please encapsulate your pattern with quotes, see -help"
+      exit 2
+fi
+
 if [ ! -z "$CHPL_HOME" ] ; then
     chapelDir="$CHPL_HOME"
 else
     chapelDir="./"
 fi
 
-pushd $chapelDir > /dev/null
+case "$1" in
+    -t)
+	type="$2"
+	shift
+	shift
+	case $type in
+	    tests)
+		filetypes=( -name \*.chpl )
+		;;
+	    futures)
+		filetypes=( -name \*.future )
+		;;
+	    perfgraphs)
+		filetypes=( -name \*.perfkeys -o -name \*.graph )
+		;;
+	    testenv)
+		filetypes=( -name \*.execenv -o -name EXECENV -o -name \*.skipif  -o -name SKIPIF -o -name \*.suppressif -o -name \*.perfexecenv -o -name PERFEXECENV )
+		;;
+	    testopts)
+		filetypes=( -name \*.compopts -o -name COMPOPTS -o -name \*.execopts  -o -name EXECOPTS -o -name \*.perfcompopts -o -name PERFCOMPOPTS -o -name \*.perfexecopts -o -name PERFEXECOPTS )
+		;;
+	    outputs)
+		filetypes=( -name \*.good -o -name \*.bad )
+		;;
+	    scripts)
+		filetypes=( -name \*.prediff -o -name PREDIFF -o -name \*.precomp  -o -name PRECOMP -o -name \*.preexec -o -name PREEXEC )
+		;;
+	    *)
+		echo "Option $type not recognized, see -help"
+		exit 3
+		;;
+	esac
+
+	if [ ${#@} -eq 0 ]
+	then
+	    echo "No search term supplied, see -help"
+	    exit 4
+	fi
+	
+	pushd $chapelDir > /dev/null
 
-find test -wholename "*/.svn" -prune -o -name \*.chpl -print0 | xargs -0 grep -H "$@"
+	find test -wholename "*/.git" -prune -o \( "${filetypes[@]}" \) -print0 | xargs -0 grep -H "$@"
 
-popd > /dev/null
+	popd > /dev/null
+	;;
+    -help)
+	echo "Usage: greptests [-t] [pattern]"
+	echo "-t [type] - the file types to search"
+	echo "    tests - .chpl"
+	echo "    futures - .future"
+	echo "    perfgraphs - .perfkeys, .graph"
+	echo "    testenv - .execenv, EXECENV, .skipif, SKIPIF, .suppressif, .perfexecenv, PERFEXECENV"
+	echo "    testopts - .compopts, COMPOPTS, .execopts, EXECOPTS, .perfcompopts, PERFCOMPOPTS, .perfexecopts, PERFEXECOPTS"
+	echo "    outputs - .good, .bad"
+	echo "    scripts - .prediff, PREDIFF, .precomp, PRECOMP, .preexec, PREEXEC"
+	echo " pattern - the text to search, please enclose any multi-word search in quotes"
```

Comment:
So with this change you can still call `greptests "blah"` but now you can additionally say `greptests -t outputs "blah"` or `greptests -t testenv "blah"`?

I think it would be good to make the PR message and final commit message be a bit more descriptive of how greptests is changed, so developers don't have to go look in the file to understand what the commit is doing.  Other than that, I like this approach

Reply:
Ok I've readded the basic `greptests "blah"` functionality, and put more information in both the final commit and the PR message